lib/private/Share20/Manager.php

```diff
@@ -1496,6 +1496,17 @@ protected function checkShare(IShare $share): void {
 				}
 			}
 		}
+
+		// For link and email shares, verify the share owner can still create such shares
+		if ($share->getShareType() === IShare::TYPE_LINK || $share->getShareType() === IShare::TYPE_EMAIL) {
+			$shareOwner = $this->userManager->get($share->getShareOwner());
+			if ($shareOwner === null) {
+				throw new ShareNotFound($this->l->t('The requested share does not exist anymore'));
+			}
+			if (!$this->userCanCreateLinkShares($shareOwner)) {
```

Comment:
```suggestion
			if (!$this->shareApiAllowLinks($shareOwner)) {
```
Please reuse the method. You can just add a parameter to specify the user and have it default to null, so it's backwards compatible.